File: 01051. Height Checker/solution.go
/**
 * Problem: 1051. Height Checker
 *
 * Difficulty: Easy
 *
 * Language: Golang
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

func heightChecker(heights []int) int {
	// Initialize a counter to keep track of mismatches between original and sorted heights
	res := 0

	// Create a new slice to store a copy of the original heights
	newArr := make([]int, len(heights))

	// Copy the contents of heights into newArr
	copy(newArr, heights)

	// Sort the copied slice to get the expected order of heights
	sort.Ints(newArr)

	// Iterate through each index to compare the original and sorted heights
	for i := range newArr {
		// If the height at the current index does not match, increment the counter
		if newArr[i] != heights[i] {
			res++
		}
	}
	
	// Return the total number of mismatches
	return res
}

File: 01051. Height Checker/solution.js
/**
 * Problem: 1051. Height Checker
 *
 * Difficulty: Easy
 *
 * Language: JavaScript
 *
 * Performance: Runtime - 0 ms (Beats 100%)
 */

/**
 * Counts indices where heights differ from their sorted order
 *
 * @param {number[]} heights - Array of heights
 *
 * @returns {number} Number of mismatched indices
 */
const heightChecker = (heights) => {
  // Create a shallow copy of the heights array to avoid mutating the original
  const sorted = [...heights]

  // Sort the copied array in ascending order
  sorted.sort((a, b) => a - b)

  // Initialize a counter for mismatched indices
  let mismatchCount = 0

  // Iterate through each index and compare original and sorted values
  for (let i = 0; i < heights.length; i++) {
    // Increment counter if values at the current index differ
    if (heights[i] !== sorted[i]) mismatchCount++
  }

  // Return the total number of mismatched indices
  return mismatchCount
}
